scanomatic/models/validators/analysis_model.py

```diff
@@ -0,0 +1,134 @@
+import os
+from typing import Literal, Union
+
+from scanomatic.data_processing.calibration import get_active_cccs
+from scanomatic.models.analysis_model import AnalysisModel, AnalysisModelFields
+from scanomatic.models.validators.tools import (
+    is_coordinates,
+    is_file,
+    is_pinning_formats,
+    is_real_number,
+    is_safe_path,
+    is_tuple_or_list
+)
+
+ValidationResult = Union[Literal[True], AnalysisModelFields]
+
+
+def validate_compilation_file(model: AnalysisModel) -> ValidationResult:
```

Comment:
and here are were the specific validators ended up, they should only have minor changes to them beyond better typing.